src/libraries/ERC7739Utils.sol

```diff
@@ -27,30 +25,30 @@ library ERC7739Utils {
     /// @param contentsHash The hash of the contents, per EIP-712
     /// @param domainBytes The encoded domain bytes from EIP-5267
     /// @param appSeparator The app's domain separator
-    /// @param contentsName The top level type, per EIP-712
-    /// @param contentsType The full type string of the contents, per EIP-712
+    /// @param contentsDescr The full type string of the contents, per EIP-712
     function toNestedTypedDataSignHash(
         bytes32 contentsHash,
         bytes memory domainBytes,
         bytes32 appSeparator,
-        string memory contentsName,
-        string memory contentsType
+        string calldata contentsDescr
     ) internal pure returns (bytes32) {
-        bytes32 typedDataSignHash = TypedDataSignLib.hash(contentsName, contentsType, contentsHash, domainBytes);
-        return MessageHashUtils.toTypedDataHash(appSeparator, typedDataSignHash);
+        (string calldata contentsName, string calldata contentsType) = decodeContentsDescr(contentsDescr);
+        if (bytes(contentsName).length == 0 || bytes(contentsType).length == 0) return bytes32(0);
+        return MessageHashUtils.toTypedDataHash(
+            appSeparator, TypedDataSignLib.hash(contentsName, contentsType, contentsHash, domainBytes)
+        );
     }
 
     /// @notice Parse the type name out of the ERC-7739 contents type description. Supports both the implicit and explicit modes
-    /// @dev Modified from https://github.com/OpenZeppelin/openzeppelin-community-contracts/blob/53f590e4f4902bee0e06e455332e3321c697ea8b/contracts/utils/cryptography/ERC7739Utils.sol
     /// @dev Returns empty strings if the contentsDescr is invalid, which must be handled by the calling function
     /// @return contentsName The type name of the contents
     /// @return contentsType The type description of the contents
-    function decodeContentsDescr(string memory contentsDescr)
+    function decodeContentsDescr(string calldata contentsDescr)
```

Comment:
this function should be directly copied from https://github.com/OpenZeppelin/openzeppelin-community-contracts/blob/master/contracts/utils/cryptography/ERC7739Utils.sol#L168